src/main/java/com/ibm/stocator/fs/common/StocatorPath.java

```diff
@@ -315,7 +267,11 @@ private String parseHadoopFOutputCommitterV1(Path fullPath,
             objName = fullPath.getName();
           }
           if (taskAttempt != null && !objName.startsWith(HADOOP_ATTEMPT)) {
-            objName = objName + "-" + taskAttempt;
+            // We want to prepend the attempt before the extension
+            String extension = extractExtension(objName);
+            objName = objName.replace("." + extension, "")
+                      + "-" + taskAttempt
+                      + "." + extension;
           }
```

Comment:
@Fokko I wonder...what happens if object doesn't have extension? Do we need: replace("." + extension, "") when extension is "" and also attach "." + extension, in case there is no extension?

Reply:
We should write a test for this, I have time later on this week.